<launch>
  <arg name="pkg_name" value="apriltag"/>
  <arg name="node_name" default="apriltag_postprocessing_node"/>
  <arg name="veh" doc="Name of vehicle. ex: csc22939"/>
  <arg name="config" default="baseline" doc="Specify a config."/>
  <arg name="param_file_name" default="default" doc="Specify a param file. ex:megaman."/>

  <group ns="$(arg veh)">
    <node name="$(arg node_name)" pkg="$(arg pkg_name)" type="$(arg node_name).py" output="screen">

        <rosparam command="load" file="$(find apriltag)/config/$(arg node_name)/$(arg param_file_name).yaml"/>
        <param name="tags_file" value="$(find apriltag)/config/apriltagsDB.yaml" />

		<!-- Remap names -->
		<remap from="apriltag_detector_node/detections" to="/$(arg veh)/apriltag_detector_node/detections"/>
		<remap from="deadreckoning/update_odom" to="/$(arg veh)/deadreckoning/update_odom"/>

    </node>
  </group>
</launch>
